### ISSUE TITLE
Support hashlock transaction

### ISSUE THREAD

**Issue by dgarcia360** [nemtech/nem2-cli#82]
# Specification

Lock a deposit needed to announce aggregate bonded transactions.

## Command

``nem2-cli transaction hashlock``

| param | type | description | default |
| -- | -- | -- | -- | 
| --profile | string | (Optional) Select between your profiles, by providing a profile  |  defaut|
| --maxfee | uint | (Optional) Maximum fee | 0 |
| --mosaicid | hex | Identifier of the mosaic being locked| |
| --amount | uint | (Optional) Amount to lock.  | 10  |
| --duration| uint |  Number of blocks for which a lock should be valid. Duration is allowed to lie up to ``2`` days | |
| --hash | string| AggregateBondedTransaction hash that has to be confirmed before unlocking the mosaics |  |

**Comment by evias** [comment]
still needed?

**Comment by dgarcia360** [comment]
Yes, it is needed. The issue is blocked because the SDK does not allow to create a HashLockTransation by hash, it requires the full signed transaction object.

**Comment by evias** [comment]
@rg911 is this something we can scope in next releases?
